### PR TITLE
Add support for sending pings to the WS server

### LINKED ISSUE #22: Support for sending pings from ws manager
Could we add functionality for sending pings from the websocket manager? I see a couple ways we can do this:

1. Run an async background process that continuously sends pings every x seconds.
2. Expose a function on the WsManager struct that sends a ping whenever it is called.

Given that 1 is what the hyperliquid-python-sdk does, it is probably the best option. To make this change non-intrusive and to not change the behavior of the WS manager for everyone, we can add this behind a feature flag.

### PR DESCRIPTION
**Pull request by sree-revoori1**
Send pings every 50 seconds so that the connections don't close.

fixes #22

### CODE REVIEW

File: src/ws/ws_manager.rs [hunk 1]
```diff
@@ -65,13 +67,22 @@ pub(crate) struct SubscriptionSendData<'a> {
...
+    const SEND_PING_INTERVAL: u64 = 50;
+
     pub(crate) async fn new(url: String) -> Result<WsManager> {
         let (ws_stream, _) = connect_async(url.clone())
             .await
             .map_err(|e| Error::Websocket(e.to_string()))?;
 
         let (writer, mut reader) = ws_stream.split();
+        let writer = Arc::new(Mutex::new(writer));
+        let writer_copy = Arc::clone(&writer);
```
Comment (lmlmt) [inline src/ws/ws_manager.rs:85]: nit: wrap the `ping_fut` block with brackets and then you can write `let writer = Arc::clone(&writer)` thereby avoiding having to write a variable suffixed with `_copy`

Comment (sree-revoori1) [reply src/ws/ws_manager.rs]: Done.

File: src/ws/ws_manager.rs [hunk 2]
```diff
@@ -88,6 +98,25 @@ impl WsManager {
         };
         spawn(reader_fut);
 
+        {
+            let writer = writer.clone();
```
Comment (lmlmt) [inline src/ws/ws_manager.rs:102]: nit: I think `let writer = Arc::clone(&writer)` is more idiomatic

**Review by lmlmt — APPROVED** [review]
Looks good, will merge once you fix merge conflict

File: src/ws/ws_manager.rs (hunk 2 above)
Comment (sree-revoori1) [reply src/ws/ws_manager.rs]: Done

**Pull request merged**